tests/kademlia/testputval.nim

```diff
@@ -0,0 +1,147 @@
+{.used.}
+import chronicles
+import strformat
+import sequtils
+import std/[times]
+# import std/enumerate
+import chronos
+import ../../libp2p/[switch, builders]
+import ../../libp2p/protocols/kademlia
+import ../../libp2p/protocols/kademlia/routingtable
+import ../../libp2p/protocols/kademlia/keys
+import ../../libp2p/protocols/kademlia/dhttypes
+import unittest2
+import ../utils/async_tests
+import std/tables
+
+proc createSwitch(): Switch =
+  SwitchBuilder
+  .new()
+  .withRng(newRng())
+  .withAddresses(@[MultiAddress.init("/ip4/0.0.0.0/tcp/0").tryGet()])
+  .withTcpTransport()
+  .withMplex()
+  .withNoise()
+  .build()
+
+proc countBucketEntries(buckets: seq[Bucket], key: Key): uint32 =
+  var res: uint32 = 0
+  for b in buckets:
+    for ent in b.peers:
+      if ent.nodeId == key:
+        res += 1
+  return res
+
+type PermissiveValidator = ref object of EntryValidator
```

Comment:
this, and selector could be moved to utilities. doc comment to indicate that usually its the responsibility of the application to define the validator and selecto